aws/rust-runtime/aws-sigv4/src/http_request/sign.rs

```diff
@@ -169,48 +169,48 @@ impl SigningInstructions {
 
     #[cfg(any(feature = "http0-compat", test))]
     /// Applies the instructions to the given `request`.
-    pub fn apply_to_request_http0x<B>(self, request: &mut http0::Request<B>) {
+    pub fn apply_to_request_http0x<B>(self, request: &mut http_02x::Request<B>) {
         let (new_headers, new_query) = self.into_parts();
         for header in new_headers.into_iter() {
-            let mut value = http0::HeaderValue::from_str(&header.value).unwrap();
+            let mut value = http_02x::HeaderValue::from_str(&header.value).unwrap();
             value.set_sensitive(header.sensitive);
             request.headers_mut().insert(header.key, value);
         }
 
         if !new_query.is_empty() {
-            let mut query = aws_smithy_http::query_writer::QueryWriter::new(request.uri());
+            let mut query = aws_smithy_http::query_writer::QueryWriter::new_from_string(
+                &request.uri().to_string(),
+            )
+            .expect("unreachable: URI is valid");
             for (name, value) in new_query {
                 query.insert(name, &value);
             }
-            *request.uri_mut() = query.build_uri();
+            *request.uri_mut() = query
+                .build_uri()
+                .to_string()
+                .parse()
```

Comment:
This looks weird. is this a 0.4 to 1.0 conversion? Why do we create a URI, stringify it, and then parse it as a URI?